diff --git a/BUILD.gn b/BUILD.gn
index 2ec17c2f07ce9384d17d0ba53010852feb013585..dfcadf78aee1e26ff485707fb830266cc2f980ef 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -189,7 +189,13 @@ config("internal_config") {
 
   include_dirs = [ "." ]
 
-  if (is_component_build) {
+  if (is_win) {
+    cflags = [
+      "/wd4275",  # dll export
+    ]
+  }
+
+  if (true) {
     defines = [ "BUILDING_V8_SHARED" ]
   }
 }
diff --git a/src/profiler/profiler-listener.cc b/src/profiler/profiler-listener.cc
index 16ec9d883a1f7fc097faf2f3f37e330b8294a5f2..c5d01e0f6ab1f03c92db6c77f969c04ba1868bfa 100644
--- a/src/profiler/profiler-listener.cc
+++ b/src/profiler/profiler-listener.cc
@@ -98,7 +98,7 @@ void ProfilerListener::CodeCreateEvent(CodeEventListener::LogEventsAndTags tag,
       int position = it.source_position().ScriptOffset();
       int line_number = script->GetLineNumber(position) + 1;
       int pc_offset = it.code_offset() + offset;
-      line_table->SetPosition(pc_offset, line_number);
+      line_table->SetPosition(pc_offset, line_number > 0 ? line_number : line);
     }
   }
   rec->entry = NewCodeEntry(
